.build/build-sonar.xml

```diff
@@ -0,0 +1,158 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<!--
+  ~ Licensed to the Apache Software Foundation (ASF) under one
+  ~ or more contributor license agreements.  See the NOTICE file
+  ~ distributed with this work for additional information
+  ~ regarding copyright ownership.  The ASF licenses this file
+  ~ to you under the Apache License, Version 2.0 (the
+  ~ "License"); you may not use this file except in compliance
+  ~ with the License.  You may obtain a copy of the License at
+  ~
+  ~     http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+<project basedir="." name="apache-cassandra-sonar-tasks" xmlns:if="ant:if" xmlns:unless="ant:unless">
+    <property name="sonar-scanner-cli.version" value="5.0.1.3006-linux"/>
+    <property name="sonar.workdir" value="${build.dir}/sonar"/>
+    <property name="sonar.download.dir" value="${user.home}/.cassandra-cache/tools"/>
+    <property name="sonar-scanner.home" value="${sonar.download.dir}/sonar-scanner-${sonar-scanner-cli.version}"/>
+    <property name="sonar.projectKey" value="cassandra"/>
+    <property name="sonar.mainBranch" value="trunk"/>
```

Comment:
This would be nice to have automatically resolved based on what branch this file is in. I get that we will probably use thing from now on but once trunk is cassandra-5.1 and we are doing some patches for it, would not it be nice if we have a way how to run Sonar against that branch too without tweaking this property every time?

Reply:
+1 to Steafan's comment. 

We have branch resolving in our shell scripts, is there any reason to keep it hardcoded in ant scripts? :-)
`branch=`git symbolic-ref -q --short HEAD 2>/dev/null || true``

Reply:
I'm not sure which branch you guys refer to.

As far as I understand, the community edition of Sonar cannot compare analysis results between branches. It can only compare to the previous analysis. In the paid developer edition, the analysis results of the current branch can be compared to the reference branch analysis results. In such case, for a feature branch, say `CASSANDRA-18785-4.1`, based on `cassandra-4.1`, `mainBranch` would be `cassandra-4.1`, and for the analysis, we would provide additionally `sonar.branch.name=CASSANDRA-18785-4.1`. Since the reference analysis results are unavailable anywhere, the developer could run the analysis manually on `cassandra-4.1` to collect the reference results and then use the same server to run the analysis on its feature branch. 

But, as said, this is a feature of the paid developer edition and is inaccessible in the community edition. Therefore, it really does not matter whether we use `trunk`, `main` (which is the default), or the current branch name for `mainBranch`. We can only make it consider a new code to be something between this and the previous run.